### PR TITLE
Check if ldap identity is compatible with account linking

### PR DESCRIPTION
**Pull request by yyorkchan**
Incoming LDAP on existing email
https://github.com/user-attachments/assets/81ee1f88-0181-40db-826d-f7c417bdd99a

Incoming LoginID on existing LDAP
https://github.com/user-attachments/assets/05571bc3-f6c9-45de-8be1-c763b0c89759

Please Note that this PR Require [#4597](https://github.com/authgear/authgear-server/pull/4597) first

### CODE REVIEW

**Comment by louischan-oursky** [comment]
@yyorkchan I have merged the dependency. Please rebase.

**Comment by yyorkchan** [comment]
Rebased Thanks @louischan-oursky

**Comment by yyorkchan** [comment]
Updated Thanks @louischan-oursky

File: e2e/tests/account_linking/incoming_oauth_conflicts_ldap.test.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      signup_flows:
+        - name: f1
+          steps:
+            - type: identify
+              one_of:
+              - identification: ldap
+              - identification: oauth
+before:
+  - type: user_import
+    user_import: ldap_user.json
```
Comment (louischan-oursky) [inline e2e/tests/account_linking/incoming_oauth_conflicts_ldap.test.yaml:32]: This import is useless. Your test relies on `ldap_user.sql` to work.

File: e2e/tests/account_linking/ldap_user.sql [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  )
+VALUES
+  (
+    '{{ $identityID }}',
+    '{{ .AppID }}',
+    'ldap-server-1',
+    'uid',
+    'mock',
+    '{"email": "mock@example.com"}',
+    '{"dn": "cn=mock,ou=people,ou=HK,dc=authgear,dc=com", "uid": "mock", "mail": "mock@example.com"}'
```
Comment (louischan-oursky) [inline e2e/tests/account_linking/ldap_user.sql:82]: This is incorrect.

1. Attributes are array, not just a string
2. Attributes are stored in bytes encoded in base64.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**